drivers/dac/dac_shell.c

```diff
@@ -94,14 +94,31 @@ static int cmd_write_value(const struct shell *sh, size_t argc, char **argv)
 	return 0;
 }
 
+static bool device_is_dac(const struct device *dev)
+{
+	return device_is_ready(dev) && DEVICE_API_IS(dac, dev);
```

Comment:
shell_device_filter already checks device_is_ready so this is superfluous

Reply:
Hey you are right! I had copy pasted this from other drivers, removed from all of these and added an extra patch to clean up the existing ones. Thanks!